### DISCUSSION TITLE
Could not find controller [...] Loading '<generic>' controller instead.

### DISCUSSION THREAD

**Discussion by omichel** [cyberbotics/webots#5391] [Q&A]
I am getting this error message, what's wrong with my robot controller?

**Comment by omichel** [comment]
This message means that Webots could neither find an executable file (e.g., `.exe`), nor an interpreted language file (e.g., `.class`, `.py`, `.m`) to run as controller program for a robot. In fact, Webots needs each controller file to be stored at specific location in order to be able to executed it. The requested location is in the "controllers" subdirectory of the current Webots project directory, e.g. "my_project". Inside the "controllers" directory, each controller project must be stored in its own directory which must be named precisely like the controller field of the [Robot](https://www.cyberbotics.com/doc/reference/robot). Inside that directory, the executable or script file must also be named after the controller field of the [Robot](https://www.cyberbotics.com/doc/reference/robot) (plus a possible extension). For example if the controller field of the robot looks like this, in the Scene Tree:

```
Robot {
  controller "my_controller"
}
```

Then the executable or script file will be searched at the following paths:

```
my_project/controllers/my_controller/my_controller.exe (Windows only)
my_project/controllers/my_controller/my_controller (Linux/Mac only)
my_project/controllers/my_controller/my_controller.class
my_project/controllers/my_controller/my_controller.py
my_project/controllers/my_controller/my_controller.m
```

If Webots does not find any file at the above specified paths, then the error message in question is shown. So this problem often happens when you:

- Have moved the project or source files to a location that does not correspond to the above description.
- Use an external build system, e.g. Visual Studio, that is not configured to generate the executable file at the right location.
- Have changed the [Robot](https://www.cyberbotics.com/doc/reference/robot)'s controller field to a location where no executable or script file can be found.
- Have reloaded the world after cleaning of the controller project.

**Reply by AlvinHenn** [reply]
Hi, if I am using C language, I need to gcc it to executable first?

**Reply by ShuffleWire** [reply]
Yes you do, C need to be built before been ran.
Please refer to documentation, there are all the useful step you might need
https://cyberbotics.com/doc/guide/using-c#introduction
